File: testing/unit-test-patterns/javascript/src/common/models/ValidationResult.ts
export class ValidationResult {
  private constructor(isValid: boolean, message = '') {
    this.isValid = isValid
    this.message = message
  }

  isValid: boolean
  message: string

  static isValidResult(): ValidationResult {
    return new ValidationResult(true)
  }

  static isInvalidResult(message: string): ValidationResult {
    return new ValidationResult(false, message)
  }
}

File: testing/unit-test-patterns/c#/UnitTestPatterns/Common/Models/ValidationResult.cs
namespace UnitTestPatterns.Common.Models
{
    public class ValidationResult
    {
        private ValidationResult(bool isValid, string message = "")
        {
            IsValid = isValid;
            Message = message;
        }

        public bool IsValid { get; }
        public string Message { get; }

        public static ValidationResult IsValidResult()
        {
            return new ValidationResult(true);
        }

        public static ValidationResult IsInvalidResult(string message)
        {
            return new ValidationResult(false, message);
        }
    }
}
